rholang/src/test/scala/coop/rchain/rholang/interpreter/LetSpec.scala

```diff
@@ -0,0 +1,126 @@
+package coop.rchain.rholang.interpreter
+
+import coop.rchain.rholang.interpreter.errors.{SyntaxError, UnexpectedProcContext}
+import org.scalatest.EitherValues._
+import org.scalatest.{Assertion, Matchers, WordSpec}
+
+/**
+  * This is the first Rholang test class introduced to test a new language construct that
+  * does not test the construct on the normalizer level. That is (1) because it is far more
+  * efficient to write and test Rholang source than to construct normalized Rholang terms
+  * corresponding to the same source, and (2) because this is the first construct that is
+  * implemented in terms of a pre-existing Rholang constructs.
+  */
+class LetSpec extends WordSpec with Matchers {
+
+  "The lexer" should {
+
+    "disallow a non-positive number of let declarations" in {
+      ParBuilderUtil.mkTerm("let in { Nil }").left.value shouldBe a[SyntaxError]
+    }
+
+    "disallow '&' and ';' from appearing in the same let declaration sequence" in {
+      ParBuilderUtil
+        .mkTerm("let x <- 1 ; y <- 2 & z <- 3 in { Nil }")
+        .left
+        .value shouldBe a[SyntaxError]
+    }
+  }
+
+  "The normalizer" should {
+
+    "treat variables declared in let processes as name variables" when {
+
+      "there is only one declaration" in {
+        ParBuilderUtil.mkTerm("let x <- 1 in { x }").left.value shouldBe a[UnexpectedProcContext]
+      }
+
+      "there are multiple sequential declarations" in {
+        ParBuilderUtil
+          .mkTerm("let x <- 1 ; y <- 2 in { x }")
+          .left
+          .value shouldBe a[UnexpectedProcContext]
+      }
+
+      "there are multiple concurrent declarations" in {
+        ParBuilderUtil
+          .mkTerm("let x <- 1 & y <- 2 in { x }")
+          .left
+          .value shouldBe a[UnexpectedProcContext]
+      }
+    }
+
+    def assertCompiledEqual(s: String, t: String): Assertion =
+      ParBuilderUtil.mkTerm(s).right.value shouldBe ParBuilderUtil.mkTerm(t).right.value
+
+    "translate a single declaration of multiple variables into a list match process" in {
+      // Variables declared in let declarations are treated as name variables,
+      // but variables bound by match expressions are treated as process variables.
+      // As long as name variables are used in name contexts and process variables are
+      // used in process contexts, the following two programs are equivalent because the normalizer
+      // eliminates the distinction between names and processes with `Par`, relying on context to
+      // decide how to process a given `p: Par`.
+      val s = "let x, @y, @{a | b} <- 1, 2, 3 | 4 in { Nil }"
```

Comment:
is this equivalent to `let x <- 1 & let @y <-2 & let @{a | b} <- 3 | 4 in { Nil }`

Reply:
Technically, no. What you write translates to a COMM rule with three consumes in a join and three produces.